File: 189-RotateArray/189-RotateArray.java
// Last updated: 6/27/2025, 2:30:19 AM
class Solution {
    public void rotate(int[] nums, int k) {
        int[] temp = new int[nums.length];
        for(int index = 0; index < nums.length; index++){
            temp[(index + k) % nums.length] = nums[index];
        }
        System.arraycopy(temp,0,nums,0,nums.length);
        System.out.println(Arrays.toString(nums));
    }
}

File: 189-RotateArray/189-RotateArray.cpp
// Last updated: 6/27/2025, 2:30:15 AM
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k = k % n; // handle if k > n
        if (k == 0) return;
        std::rotate(nums.begin(), nums.begin() + (n - k), nums.end());
    }
};